src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
I'm still struggling to understand precisely what problem it is the purpose of these changes to fix.

I'm unconvinced that the change to ```integers()``` strategy is not a first step in the wrong direction.

I'm going to ask my former professor, retired, about the problem of deciding whether there exists a rational number of a designated denominator between two rational bounds. He's a an algebraist, he might be interested. But that's orthogonal to the main problem, AFAIAC.